core/store/migrations/0020_remove_result_task.go

```diff
@@ -0,0 +1,42 @@
+package migrations
+
+import (
+	"github.com/go-gormigrate/gormigrate/v2"
+	"gorm.io/gorm"
+)
+
+const (
+	up20 = `
+		ALTER TABLE pipeline_task_runs DROP CONSTRAINT chk_pipeline_task_run_fsm;
+		ALTER TABLE pipeline_task_runs 
+    		ADD CONSTRAINT chk_pipeline_task_run_fsm CHECK (
+				((finished_at IS NOT NULL) AND (num_nonnulls(output, error) != 2))
```

Comment:
Presumably we don't want to allow just one of these to be null and the other not null though?

Reply:
that is what we want for pipeline_task_runs - maybe you're thinking of pipeline_runs where we want them to both not be null? In any event I just kept the logic the same as the old constraint